### ISSUE TITLE
Query parsing breaks with encoded equals sign

### ISSUE THREAD

**Issue by philipstears** [freya-fs/arachne#6]
The following code results in a crash
```fsharp
    let queryParsed = Query.Parse "one=two%3d"
    let queryPairs = queryParsed |> (fst <| Query.Pairs_)
```

I believe the issue is that percent decoding is being performed too early, leading to the parser thinking that the %3d is actually a key/value separator in the query.

**Comment by kolektiv** [comment]
Yes, you're right. For Arachne 3.0 I'm going to have to reconsider how we deal with encoding, as this (and similar issues) are causing some odd edge cases.

**Comment by kolektiv** [comment]
I think this should be fixed by #25 but I want to check it before the 3.0 release.

**Comment by kolektiv** [comment]
In fact this was not entirely fixed by that and exposed a bug in the percent-encoding logic - it didn't cope well with lowercase encodings! That will now be fixed for 3.0. I think this does imply some further design work needed around query and pairs optics, but that can be done post-3.0.

**Comment by kolektiv** [comment]
Fixed in PR #35
